doc/source/building/index.rst

```diff
@@ -0,0 +1,354 @@
+      git clone https://github.com/scipy/scipy.git
+      git submodule update --init
+      pip install . --no-build-isolation
+
+
+.. _the-dev-py-interface:
+
+Building from source for SciPy development
+``````````````````````````````````````````
+
+If you want to build from source in order to work on SciPy itself, first clone
+the SciPy repository::
+
+      git clone https://github.com/scipy/scipy.git
+      git submodule update --init
+
+Then you want to do the following:
+
+1. Create a dedicated development environment (virtual environment or conda
+   environment),
+2. Install all needed dependencies (*build*, and also *test*, *doc* and
+   *optional* dependencies), 
+3. Build SciPy with our ``dev.py`` developer interface.
+
+Step (3) is always the same, steps (1) and (2) are different between conda and
+virtual environments:
+
+.. tab-set::
+
+  .. tab-item:: Virtual env or system Python
+
+    .. note::
+
+       There are many tools to manage virtual environments, like ``venv``,
+       ``virtualenv``/``virtualenvwrapper, ``pyenv``/``pyenv-virtualenv``,
+       Poetry, PDM, Hatch, and more. Here we use the basic ``venv`` tool that
+       is part of the Python stdlib. You can use any other tool; all we need is
+       an activated Python environment.
+
+    Create and activate a virtual environment in a new directory named ``venv``::
+
+       python -m venv venv
+       source venv/bin/activate
+
+    Then install the Python-level dependencies (see ``pyproject.toml``) from
+    PyPI with::
+
+       # Build dependencies
+       python -m pip install numpy cython pythran pybind11 meson ninja pydevtool rich-click
+
+       # Test and optional runtime dependencies
+       python -m pip install pytest pytest-xdist pytest-timeout pooch threadpoolctl asv gmpy2 mpmath
+
+       # Doc build dependencies
+       python -m pip sphinx "pydata-sphinx-theme==0.9.0" sphinx-design matplotlib numpydoc jupytext myst-nb
+
+       # Dev dependencies (static typing and linting)
+       python -m pip mypy typing_extensions types-psutil pycodestyle ruff cython-lint
+
+  .. tab-item:: Conda env
+
+    If you don't have a conda installation yet, we recommend using
+    Mambaforge_; any conda flavor will work though.
+
+    To create a ``scipy-dev`` development environment with every required and
+    optional dependency installed, run::
+
+        mamba env create -f environment.yml
+        mamba activate scipy-dev
+
+    .. note::
+
+       On Windows it is possible that the environment creation will not work due
+       to an outdated Fortran compiler. If that happens, remove the ``compilers``
+       entry from ``environment.yml`` and try again. The Fortran compiler should
+       be installed as described under the *Windows* tab of the *System-level
+       dependencies* section higher up.
+
+To build SciPy in an activated development environment, run::
+
+    python dev.py build
```

Comment:
IIUC, gh-18451 should make this reliable enough to recommend without `-jX`.👍

Reply:
indeed.